### PR TITLE
mempool: remove vestigal mempool wal

### PR DESCRIPTION
**Pull request by tychoish**
I can't find any code that's actually *reading* for the WAL in the
mempool, and writing mempool transactions to disk seems unexpected at best.

### CODE REVIEW

**Review by cmwaters — APPROVED** [review]
I must admit it's a bit peculiar to have a WAL for something in-memory. I'm all for removing seemingly unused code but I would defer to @alexanderbez on whether we should be doing this. If he's okay and the linter is okay then I approve this 👍

**Comment by tessr** [comment]
I'd also feel better about any WAL changes if @ebuchman could give a cursory thumbs up. I know this stuff has been finicky in the past.

**Review by alexanderbez — APPROVED** [review]
I do not see any existing replay logic so this seems reasonable to me 👍

**Comment by marbar3778** [comment]
The initial idea was similar to the consensus wal. You write to the wal in case of failure in order to repopulate the mempool. The second part, reading after failure never got implemented.

**Review by ebuchman — COMMENTED** [review]
So I think this is fine, this was never actually used programmatically, it's just there as a stop gap. But I don't know if there's any users that might rely on it for instance with some external script to replay mempool msgs after a restart. The goal was to provide some kind of stronger guarantee that txs would not get lost even before they are committed but it's hard to say if/how anyone depends on it.

It may still be a good idea for certain say more enterprise use cases but given the whole mempool will be redesigned we can leave it for that.

Love me some nice code deletion ;)

**Comment by tychoish** [comment]
> So I think this is fine, this was never actually used programmatically, it's just there as a stop gap. But I don't know if there's any users that might rely on it for instance with some external script to replay mempool msgs after a restart. The goal was to provide some kind of stronger guarantee that txs would not get lost even before they are committed but it's hard to say if/how anyone depends on it.

I'll add an item to the change log just to be safe? This doesn't feel super well supported, and is certainly not a thing we test (you probably wouldn't want to do this between even minor versions of tendermint.) 

> It may still be a good idea for certain say more enterprise use cases but given the whole mempool will be redesigned we can leave it for that.

legit. I think having a persistent uncommitted transaction buffer (e.g. mempool, "persistent mempool" being a bit of a peculiarity as a term ) is a reasonable feature to add, but I think it makes sense to implement that distinctly from this code.

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ Friendly reminder: We have a [bug bounty program](https://hackerone.com/tendermi
 
 - Data Storage
   - [store/state/evidence/light] \#5771 Use an order-preserving varint key encoding (@cmwaters)
+  - [mempool] \#6396 Remove mempool's write ahead log (WALL), (previously unused by the tendermint code). (@tychoish)
```
Comment (tessr) [inline CHANGELOG_PENDING.md:55]:
```suggestion
  - [mempool] \#6396 Remove mempool's write ahead log (WAL), (previously unused by the tendermint code). (@tychoish)
```

**Pull request merged**